instrumentation-core/spring/spring-webflux-5.0/src/main/java/io/opentelemetry/instrumentation/springwebflux/client/WebClientTracingFilter.java

```diff
@@ -49,32 +48,27 @@ public static void addFilter(
 
   @Override
   public Mono<ClientResponse> filter(final ClientRequest request, final ExchangeFunction next) {
-    Span span = DECORATE.getOrCreateSpan(request, tracer);
-    DECORATE.afterStart(span);
+    Span span = TRACER.getOrCreateSpan(TRACER.spanNameForRequest(request), tracer);
 
     try (Scope scope = tracer.withSpan(span)) {
       ClientRequest mutatedRequest =
           ClientRequest.from(request)
-              .headers(httpHeaders -> DECORATE.inject(Context.current(), httpHeaders))
+              .headers(httpHeaders -> TRACER.inject(Context.current(), httpHeaders))
               .build();
-      DECORATE.onRequest(span, mutatedRequest);
+      TRACER.onRequest(span, mutatedRequest);
```

Comment:
this can go up, before the scope, which helps us see it's all part of "startSpan"